bindings/go/runtime/identity_match.go

```diff
@@ -0,0 +1,93 @@
+package runtime
+
+import (
+	"maps"
+	"path"
+)
+
+// IdentityMatchingFn is a function that takes two identities and returns if they match.
+// It is expected that the function can modify the identities in place.
+// If a comparison is absolute, the function can choose to delete attributes.
```

Comment:
This is not a good enough description for me, sorry. :D Why does an IDENTITY MATCH have to delete attributes or _could_? This is like saying, `Equal` might delete some keys while it's checking. I'm not sure why this is a possibility. This seems like an error waiting to happen.

Reply:
Explained eeven further. You conveniently left out the last comment line that was also hinting at the chaining :D